config/internal/apiserver/default/deployment.yaml.tmpl

```diff
@@ -250,7 +254,14 @@ spec:
             - --https-address=:8443
             - --provider=openshift
             - --openshift-service-account={{.APIServerDefaultResourceName}}
+            {{ if .InterPodTLS }}
+            # because we use certs signed by openshift, these certs are not valid for
+            # localhost, thus we have to use the service name
+            - --upstream=https://{{.APIServerServiceDNSName}}:8888
```

Comment:
I wonder what this name resolves to. Hopefully some IP that points directly at one of the pods (or the only pod) backing the server, and not some type of haproxy router. Hmm. If it was some type of router, I think about traffic leaving the node ... but it is over TLS at this point so ... wouldn't matter?

Reply:
it will resolve to something like: 

```
- '--upstream=https://ds-pipeline-pr-40.dspa1.svc.cluster.local:8888'
```
note that this is pointing to the side car apiserver container within the same pod, this container is now tls enabled with this configuration

unfortunately we can't use IPs or localhost, because then you'll get an IP SANs error